src/shared/services/auth/auth.ts

```diff
@@ -146,15 +148,20 @@ function catchError() {
  * null otherwise.
  */
 function getAccessTokenFromParams(params: { [key: string]: string }) {
-  if (!params || typeof window === 'undefined') {
+  /* istanbul ignore next */
+  if (typeof window === 'undefined') {
+    return null
+  }
+
+  if (!params) {
     return null
   }
 
   const stateToken = sessionStorage.getItem(STATE_TOKEN)
 
   // The state param must be exactly the same as the state token we
   // have saved in the session (to prevent CSRF)
-  const stateTokenValid = params.state && params.state === stateToken
+  const stateTokenValid = Boolean(params.state && params.state === stateToken)
```

Comment:
I think we can actually clean this up with some optional chaining:
```suggestion
  const stateTokenValid = params?.state === stateToken
```